### PR TITLE
fix(Schedule Trigger Node): Change scheduler behaviour for intervals days and hours

### PR DESCRIPTION
**Pull request by agobrech**
Github issue / Community forum post (link here to close automatically):
https://community.n8n.io/t/confused-re-schedule-node-executing-at-the-same-time-every-day/20324?u=mutedjam

### CODE REVIEW

File: packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts [hunk 1]
```diff
@@ -485,10 +511,23 @@ export class ScheduleTrigger implements INodeType {
...
 				const minute = interval[i].triggerAtMinute?.toString() as string;
 				const cronTimes: string[] = [minute, `*/${hour}`, '*', '*', '*'];
 				const cronExpression: string = cronTimes.join(' ');
+				if (hour === 1) {
+					const cronJob = new CronJob(cronExpression, executeTrigger, undefined, true, timezone);
+					cronJobs.push(cronJob);
+				} else {
+					const cronJob = new CronJob(
+						cronExpression,
+						() => executeTrigger(i, hour, 'hours'),
```
Comment (krynble) [inline packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts:524]: It's starting to get confusing having these as optional information. We should make this mandatory and change all recurrency rules to send this information and adapt the code to use it whenever needed.

File: packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts [hunk 2]
```diff
@@ -485,10 +511,23 @@ export class ScheduleTrigger implements INodeType {
 			}
 
 			if (interval[i].field === 'hours') {
-				const hour = interval[i].hoursInterval?.toString() as string;
+				const hour = interval[i].hoursInterval as number;
 				const minute = interval[i].triggerAtMinute?.toString() as string;
 				const cronTimes: string[] = [minute, `*/${hour}`, '*', '*', '*'];
 				const cronExpression: string = cronTimes.join(' ');
+				if (hour === 1) {
```
Comment (krynble) [inline packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts:518]: I think we're now relying too much on the Cron lib and for an execution that should happen every X hours the `setInterval` would be a better approach, WDYT?

File: packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts [hunk 3]
```diff
@@ -435,22 +439,44 @@ export class ScheduleTrigger implements INodeType {
...
-					moment.tz(timezone).week() + 52 >= weeksInterval + lastExecutionWeekNumber // not first time, correct interval but year has passed
+					lastExecution === undefined || // First time executing this rule
+					moment.tz(timezone).week() >= intervalSize + lastExecution || // not first time, but minimum interval has passed
+					moment.tz(timezone).week() + 52 >= intervalSize + lastExecution // not first time, correct interval but year has passed
 				) {
 					staticData.recurrencyRules[recurrencyRuleIndex] = moment.tz(timezone).week();
 					this.emit([this.helpers.returnJsonArray([resultData])]);
 				}
 				// There is no else block here since we don't want to emit anything now
+			} else if (intervalSize && recurrencyRuleIndex !== undefined && typeInterval === 'days') {
```
Comment (krynble) [inline packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts:462]: This is starting to become a bit spaghetti - maybe refactor to take into accounts scenarios based on the recurrence rule?

**Review by krynble — DISMISSED** [review]

**Review by maspio — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.214.0`

**Pull request merged**